### PR TITLE
feat: SecretGenerator and CertificateRenewer

### PR DESCRIPTION
**Pull request by samuel-hawker**
### Type of change
- Refactoring

### Description

Split out utility methods to their own classes
for better encapsulation and readability

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by ppatierno** [comment]
Guess that this PR suffers the same "problems" of the `ProbeGenerator` one?

**Comment by samuel-hawker** [comment]
I have a plan in tackling the call patterns reverting them back to static methods - aside from the call pattern issue, does this look like a welcome change?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CertificateRenewer.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+     * @param labels the labels te secret should contain
+     * @param ownerReference the OwnerReference for the secret
+     */
+    public Secret signedCertificateSecret(Secret secret, String secretName, String namespace, Labels labels, OwnerReference ownerReference) {
+        Map<String, String> signedCertData = new HashMap<>(4);
+        CertAndKey certAndKey = null;
+
+        Optional<CertAndKey> regeneratedCert = maybeRegenerateCertificate(clusterCa, secret, commonName);
+
+        // TODO ensure empty check is fine
```
Comment (samuel-hawker) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CertificateRenewer.java:68]: This is slightly differed from the original logic, all tests still pass but in the case where a cert is supposed to be generated but isnt due to an io exception, we now pass through into the else, does this make sense?

Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CertificateRenewer.java:73]: @ppatierno @tombentley I think you two worked on this IIRC. Any clue about ^^^?

Comment (tombentley) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CertificateRenewer.java:73]: Well why is it OK to ignore the IOE when generating a cert? It seems to me that conflating "cert not required" and "cert required but could not be generated due to some error" is likely to result in the cert just quietly never getting regenerated in the error case.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/SecretGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+
+class SecretGeneratorTest {
+
+    final static String NAME = "secret-name";
+    final static String NAMESPACE = "my-namespace";
+    final static Labels LABELS = Labels.EMPTY.withStrimziName("component");
```
Comment (see-quick) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/SecretGeneratorTest.java:26]: You do not need to use `static` keyword or do you also using these instance variables inside @AfterAll or @BeforeAll phases?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/CertificateRenewerTest.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class CertificateRenewerTest {
+
+    final static String SECRET_NAME = "secret-name";
+    final static String NAMESPACE = "my-namespace";
+    final static String COMPONENT_COMMON_NAME = "my-component-common-name";
+    final static String KEY_CERT_NAME = "my-component-key-cert-name";
```
Comment (see-quick) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/CertificateRenewerTest.java:32]: Same here

**Comment by samuel-hawker** [comment]
Closing to rework this. Will re-open later.

**Pull request closed without merging**